### ISSUE TITLE
WalletConect widget popping out repeatedly

### ISSUE THREAD

**Issue by tomislavmamic** [Giveth/giveth-dapps-v2#4109]
I signed in with my Metamask and got this behaviour immediately: https://cln.sh/r5P290nR

**Comment by maryjaf** [comment]
Thanks for reporting
When the user selects "Wallet connect" option for sign in flow, then cancel it, this modal won't be closed 
please take a look. @MohammadPCh 


https://github.com/Giveth/giveth-dapps-v2/assets/111529185/49106021-bc8a-4ffa-bdeb-f7d38644872a

https://github.com/Giveth/giveth-dapps-v2/assets/111529185/8bf2ed9a-51d5-45e6-841f-bf8380b07a93

**Comment by kkatusic** [comment]
@MohammadPCh I investigated this problem, it occurs like @maryjaf said. Tried on empty page with only one connect button option and still get same problem, tried wit newest version of the @web3modal/wagmi package, but same problem. Looked at issues at package @web3modal/wagmi, wagmi, @walletconnect there isn't any type off double window showing. 

I guess that web3modal use wallet connect options for generating its qr code and for some reason it trigger also wallet connect modal window.

Maybe there is some hydration problem related to nextjs flow, I don't know.

@maryjaf hot fix is on this [branch](https://github.com/Giveth/giveth-dapps-v2/tree/fix_wallet_connect_modal)
